rs/ledger_suite/common/ledger_core/src/timestamp.rs

```diff
@@ -66,3 +68,20 @@ impl Sub<Duration> for TimeStamp {
         }
     }
 }
+
+impl Storable for TimeStamp {
+    fn to_bytes(&self) -> Cow<[u8]> {
+        Cow::Owned(self.as_nanos_since_unix_epoch().to_le_bytes().to_vec())
```

Comment:
Maybe we could have a test that `as_nanos_since_unix_epoch` returns 64-bit integer?

Reply:
You mean a test that makes sure that the fn return type does not change? I added a proptest for serialization: https://github.com/dfinity/ic/pull/1513/commits/86656951223d47befb76afd5d3c13db566bdc0b6

Reply:
The proptest won't catch regressions if the return type changes across commits.

Reply:
I am not sure I understand, I tried:
```
    pub fn as_nanos_since_unix_epoch(&self) -> u128 {
        self.timestamp_nanos as u128
    }
```
and similarly with `u32` and the test fails...

Reply:
I missed that the function is [defined](https://github.com/dfinity/ic/blob/c57a842fd06dea5e95ab608fb00cca3c921c3da1/rs/ledger_suite/common/ledger_core/src/timestamp.rs#L28) in your code so you can make sure the return type does not change - I though it is a library dependency that could silently change across a version bump. So nevermind my comment.